File: 4011-count-subarrays-with-even-odd-ratio-i/4011-count-subarrays-with-even-odd-ratio-i.cpp
class Solution {
public:
    int countRatioSubarrays(vector<int>& nums, int a, int b) {
        int i, j, c = 0, l = nums.size();
        for (i = 0; i < l; i++) {
            int o = 0, e = 0;
            for (j = i; j < l; j++) {
                if (nums[j] % 2 == 0)
                    e++;
                else
                    o++;
                if (o > 0 && e * b <= a * o)
                    c++;
            }
        }
        return c;
    }
};

File: 4011-count-subarrays-with-even-odd-ratio-i/4011-count-subarrays-with-even-odd-ratio-i.py
class Solution(object):
    def countRatioSubarrays(self, nums, a, b):
        """
        :type nums: List[int]
        :type a: int
        :type b: int
        :rtype: int
        """
        n=len(nums)
        count=0
        for i in range(n):
            e=0
            o=0
            for j in range(i,n):
                if (nums[j]%2==0):
                    e+=1
                else:
                    o+=1
                if (o>0 and e * b<=o* a):
                    count+=1
        return count
        